code/modules/reagents/reagent_containers/cups/_cup.dm

```diff
@@ -65,152 +64,90 @@
 
-		if(reagents.holder_full())
-			to_chat(user, span_warning("[src] is full."))
-			return ITEM_INTERACT_BLOCKING
-
-		var/trans = round(target.reagents.trans_to(src, amount_per_transfer_from_this, transferred_by = user), CHEMICAL_VOLUME_ROUNDING)
-		playsound(target.loc, SFX_LIQUID_POUR, 50, TRUE)
-		to_chat(user, span_notice("You fill [src] with [trans] unit\s of the contents of [target]."))
-		SEND_SIGNAL(src, COMSIG_REAGENTS_CUP_TRANSFER_FROM, target)
-		target.update_appearance()
-		return ITEM_INTERACT_SUCCESS
+	if(isliving(target))
+		return try_drink(target, user)
 
 	return NONE
 
 /obj/item/reagent_containers/cup/interact_with_atom_secondary(atom/target, mob/living/user, list/modifiers)
-	if(user.combat_mode)
-		return NONE
-	if(!check_allowed_items(target, target_self = TRUE))
-		return NONE
-	if(!spillable)
+	. = ..()
+	if(. & ITEM_INTERACT_ANY_BLOCKER)
+		return .
+	if(!is_open_container())
 		return NONE
 
 	if(target.is_drainable()) //A dispenser. Transfer FROM it TO us.
-		if(!target.reagents.total_volume)
-			to_chat(user, span_warning("[target] is empty!"))
-			return ITEM_INTERACT_BLOCKING
-
-		if(reagents.holder_full())
-			to_chat(user, span_warning("[src] is full."))
-			return ITEM_INTERACT_BLOCKING
-
-		var/trans = round(target.reagents.trans_to(src, amount_per_transfer_from_this, transferred_by = user), CHEMICAL_VOLUME_ROUNDING)
-		playsound(target.loc, SFX_LIQUID_POUR, 50, TRUE)
-		to_chat(user, span_notice("You fill [src] with [trans] unit\s of the contents of [target]."))
-		SEND_SIGNAL(src, COMSIG_REAGENTS_CUP_TRANSFER_FROM, target)
-		target.update_appearance()
-		return ITEM_INTERACT_SUCCESS
+		return try_drain(target, user)
 
 	return NONE
 
-/obj/item/reagent_containers/cup/attackby(obj/item/attacking_item, mob/user, list/modifiers, list/attack_modifiers)
-	var/hotness = attacking_item.get_temperature()
-	if(hotness && reagents)
-		reagents.expose_temperature(hotness)
-		to_chat(user, span_notice("You heat [name] with [attacking_item]!"))
-		return TRUE
-
-	//Cooling method
-	if(istype(attacking_item, /obj/item/extinguisher))
-		var/obj/item/extinguisher/extinguisher = attacking_item
-		if(extinguisher.safety)
-			return TRUE
-		if (extinguisher.reagents.total_volume < 1)
-			to_chat(user, span_warning("\The [extinguisher] is empty!"))
-			return TRUE
-		var/cooling = (0 - reagents.chem_temp) * extinguisher.cooling_power * 2
-		reagents.expose_temperature(cooling)
-		to_chat(user, span_notice("You cool \the [src] with the [attacking_item]!"))
-		playsound(loc, 'sound/effects/extinguish.ogg', 75, TRUE, -3)
-		extinguisher.reagents.remove_all(1)
-		return TRUE
-
-	if(istype(attacking_item, /obj/item/food/egg)) //breaking eggs
-		var/obj/item/food/egg/attacking_egg = attacking_item
-		if(!reagents)
-			return TRUE
+/obj/item/reagent_containers/cup/item_interaction(mob/living/user, obj/item/tool, list/modifiers)
+	if(!is_open_container())
+		return NONE
+	if(istype(tool, /obj/item/food/egg)) //breaking eggs
```

Comment:
Not your fault but god this should totally be on eggs, I should probably fix it later